source/pem_utils.c

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+#include <aws/common/string.h>
+#include <aws/io/private/pem_utils.h>
+
+enum aws_pem_util_state {
+    BEGIN,
+    ON_DATA,
+    END,
+};
+
+int aws_clean_up_pem(struct aws_byte_buf *pem, struct aws_allocator *allocator) {
+    if (!pem->len) {
+        return AWS_OP_SUCCESS;
+    }
+    char *clean_pem = NULL;
+    clean_pem = aws_mem_calloc(allocator, pem->len, sizeof(char));
+    if (!clean_pem) {
+        return AWS_OP_ERR;
+    }
+    struct aws_byte_cursor pem_cursor = aws_byte_cursor_from_buf(pem);
+    int state = BEGIN;
+    int clean_pem_index = 0;
+    const char *begin_header = "-----BEGIN";
```

Comment:
Can we do this with byte cursors and aws_strings?

Reply:
you mean the `begin_header` to compare, or the clean_pem we are writing into? I basically follow the [pki_utils](https://github.com/awslabs/aws-c-io/blob/main/source/pki_utils.c#L43)😂

Reply:
I think he was suggesting using `AWS_BYTE_CUR_INIT_FROM_STRING_LITERAL` and using byte cursor comparisons